python/Sailbot.py

```diff
@@ -121,8 +126,28 @@ def getCurrentState(self):
             rospy.logwarn("Setting globalWaypoint to be the last element of the globalPath")
             self.globalPathIndex = len(self.globalPath) - 1
 
-        return BoatState(self.globalPath[self.globalPathIndex], self.position, self.measuredWindDirectionDegrees,
-                         self.measuredWindSpeedKmph, self.AISData, self.headingDegrees, self.speedKmph)
+        state = BoatState(self.globalPath[self.globalPathIndex], self.position, self.measuredWindDirectionDegrees,
+                          self.measuredWindSpeedKmph, self.AISData, self.headingDegrees, self.speedKmph)
+
+        global goalWasInvalid
+        global movedGlobalWaypoint
+        if goalWasInvalid or not his.checkGoalValidity(state):
```

Comment:
if not goalwasinvalid and not checkgoalvalidity:
   goalwasinvalid = true

if goalwasinvalid
   state.globalwaypoint = movedglobalwaypoint